src/main/java/duke/MainWindowController.java

```diff
@@ -0,0 +1,373 @@
+        assert masterPane != null : "fx:id=\"masterPane\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert masterListView != null :
+                "fx:id=\"masterListView\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert detailPane != null : "fx:id=\"detailPane\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert middleListView != null :
+                "fx:id=\"middleListView\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert detailLabel != null : "fx:id=\"detailLabel\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert detailTypeLabel != null :
+                "fx:id=\"detailTypeLabel\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert detailDescriptionLabel != null :
+                "fx:id=\"detailDescriptionLabel\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert detailTimeLabel != null :
+                "fx:id=\"detailTimeLabel\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert detailTypeText != null :
+                "fx:id=\"detailTypeText\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert detailDescriptionTextArea != null :
+                "fx:id=\"detailDescriptionTextArea\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert detailTimeText != null :
+                "fx:id=\"detailTimeText\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert sortMenu != null : "fx:id=\"sortMenu\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert sortItemByNameAscend != null :
+                "fx:id=\"sortItemByNameAscend\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert sortItemByNameDescend != null :
+                "fx:id=\"sortItemByNameDescend\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert sortItemByTimeAscend != null :
+                "fx:id=\"sortItemByTimeAscend\" was not injected: check your FXML file 'MainWindow.fxml'.";
+        assert sortItemByTimeDescend != null :
+                "fx:id=\"sortItemByTimeDescend\" was not injected: check your FXML file 'MainWindow.fxml'.";
+
+        setDetailVisible(false);
+        hideSortByTime();
+        middleListView.setCellFactory(param -> {
+
+            ListCell<Task> cell = new CustomListCell();
+            // MenuItem for delete task
+            MenuItem deleteItem = new MenuItem();
+            deleteItem.setText("Delete");
+            deleteItem.setOnAction(event -> {
+                duke.getTasks().remove(cell.getItem());
+                try {
+                    duke.updateStorage();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                refreshView();
+            });
```

Comment:
Perhaps you can contain this whole logic block under a separate function (e.g. "getMenuItemForDelete", "getMenuItemForDone")